### PR TITLE
Fix (brevitas/quantizer): fixed key-error when combining static quantization and accelerate with CPU offload

### PR DESCRIPTION
**Pull request by nickfraser**
Note, this fix relies on [this fix](https://github.com/huggingface/accelerate/pull/2475) in accelerate.

### CODE REVIEW

File: optimum/amd/brevitas/quantizer.py [hunk 1]
```diff
@@ -206,7 +206,10 @@ def quantize(
         # https://github.com/Xilinx/brevitas/blob/84f42259ec869eb151af4cb8a8b23ad925f493db/src/brevitas/core/scaling/standalone.py#L205-L217
         with torch.no_grad():
             if calibration_dataset is not None:
+                is_training = model.training
+                model.train()
                 model(**calibration_dataset[0])
+                model.train(is_training)
```
Comment (fxmarty) [inline optimum/amd/brevitas/quantizer.py:212]: @nickfraser can you explain the reasoning?

Comment (Giuseppe5) [reply optimum/amd/brevitas/quantizer.py:212]: It's similar to the reason why we need a CPU forward pass in the first place.

For activation quantization and PTQ, we assume that scale factors are computed through statistics over the input data.
If you try to retrieve the state_dict of the model before any of these statistics was actually computed, then we don't include that value in the state_dict as it would be meaningless. However this conflicts with accelerate.
For weights scale factors, we just need a forward pass (and it does not matter if the model is in training or eval mode). For activations, statistics are computed only in training mode.

Comment (fxmarty) [reply optimum/amd/brevitas/quantizer.py:212]: > For activations, statistics are computed only in training mode.

Why?

**Comment by Giuseppe5** [comment]
I might have a different solution for this which would allow to skip the CPU forward pass entirely. We need to test it first though

File: optimum/amd/brevitas/quantizer.py (hunk 1 above)
Comment (fxmarty) [reply optimum/amd/brevitas/quantizer.py:212]: I see, makes sense.

Does this mean that during calibration the model is in train mode (there's no explicit `model.train()` in here, but maybe in brevitas)? Does that not mess up e.g. if dropout is used?

Comment (Giuseppe5) [reply optimum/amd/brevitas/quantizer.py:212]: The `calibration_mode` context manager is used for the purpose of setting in train mode only the part of the networks that need to be in train mode, while leaving everything else in eval mode.

Comment (nickfraser) [reply optimum/amd/brevitas/quantizer.py:212]: This issue only surfaced under the following conditions:
 - `--is-static` flag is set
 - `--device auto`
 - The model was big enough that it didn't fit entirely in a single GPU (accelerate needed to spill to offload)

The fix avoids this issue.

**Comment by nickfraser** [comment]
Worse-case, this will temporarily work around the issue under the these conditions:

>This issue only surfaced under the following conditions:
> - `--is-static` flag is set
> - `--device auto`
> - The model was big enough that it didn't fit entirely in a single GPU (accelerate needed to spill to offload)

**Comment by nickfraser** [comment]
Superseeded by #87.
